lib/perl/Genome/SoftwareResult/User.t

```diff
@@ -68,6 +72,7 @@ my $new_sr = $class->with_registered_users(
             callback => \&_newly_created_callback,
             %users_hash
         );
+UR::Context->commit();
```

Comment:
If two things in the same process are trying to get_or_create the same user, the second one should find the one created by the first one in the `get` before locking.

The observers are firing or we would not successfully find the user relationships in the test.  (For what it's worth we had STDERR prints in the second observer originally to make sure it fired.)  I thought `UR::Context->commit();` was on the process if you're not in a transaction.